src/Microsoft.OData.Core/Uri/ODataUri.cs

```diff
@@ -214,7 +214,23 @@ public Uri ServiceRoot
         /// <summary>
         /// Gets or sets the ParameterAliasValueAccessor.
         /// </summary>
-        internal ParameterAliasValueAccessor ParameterAliasValueAccessor { get; set; }
+        internal ParameterAliasValueAccessor ParameterAliasValueAccessor
+        {
+            get
+            {
+                if (parameterAliasValueAccessor == null)

```

Comment:
This is a change of behaviour - could be for the better, but still a change :-) - and there probably should be a test to expect this current behaviour...
- Previously: value is set in ctor, you could then be able to `set` a null to that value and a subsequent `get` would give null.
- Currently: even if you ever `set` a null value, you would never `get` it
It could be that current behaviour is fixing unwanted null gets but yeah, probably a test to document this...

Reply:
Hmm, good point

Reply:
Would it be possible to change the setter from internal to private ?

Reply:
Even the getter, does not seem it needs to be public as of now.

Reply:
Yes; the setter is currently internal and only called when initialized to a non-null value in the constructor, so we should be able to assume it is never null.  We can change the setter (and getter) to private and change the null check on 128-133 with an non-null assert.